2009-08-22  Peter Breitenlohner  <peb@mppmu.mpg.de>

	* vlna.w (normalpattern): Constify string arg.
	Drop unused global 'pi'.

diff -ur vlna-1.5.orig/vlna.w vlna-1.5/vlna.w
--- vlna-1.5.orig/vlna.w	2011-05-02 15:58:47.000000000 +0200
+++ vlna-1.5/vlna.w	2011-05-02 16:00:12.000000000 +0200
@@ -380,7 +380,7 @@
 PATITEM *lapi[MAXPATT];      /* pole ukazatelu na aktualni pozice */
 PATTERN *lapt[MAXPATT];      /* pole odpovidajicich ukazatelu na patterny */
 PATTERN *listpatt, *normallist, *commentlist, *pt, *lastpt=NULL;
-PATITEM *pi, *lastpi=NULL;
+PATITEM *lastpi=NULL;
 char c;             /* zrovna nacetny znak */
 char buff[MAXBUFF]; /* prechodny buffer */
 int ind;            /* aktualni pozice prechodneho bufferu */
@@ -472,7 +472,7 @@
 
 @ Definujme funkci |normalpattern|.
 @<Pomocné funkce@>=
-static PATTERN *normalpattern (void (*proc)(void), char *str)
+static PATTERN *normalpattern (void (*proc)(void), const char *str)
 {
   PATTERN *pp;
   int j=0;
